www/src/pages/de/usage/next-auth.md

```diff
@@ -0,0 +1,176 @@
+---
+title: NextAuth.js
+description: Verwendung von NextAuth.js
+layout: ../../../layouts/docs.astro
+lang: de
+---
+
+Wenn du ein Authentifizierungssystem in deiner Next.js-Anwendung haben möchtest, ist NextAuth.js eine ausgezeichnete Lösung, um die Komplexität zu vermeiden, es selbst zu bauen. Es kommt mit einer umfangreichen Liste von Providern, um OAuth-Authentifizierung schnell hinzuzufügen und bietet Adapter für viele Datenbanken und ORMs.
+
+## Context Provider
+
+In den Einstiegspunkt deiner Anwendung, wirst du sehen, dass deine Anwendung von einem [SessionProvider](https://next-auth.js.org/getting-started/client#sessionprovider) umschlossen wird.
+
+```tsx:pages/_app.tsx
+<SessionProvider session={session}>
+  <Component {...pageProps} />
+</SessionProvider>
+```
+
+Dieser Context-Provider ermöglicht es deiner Anwendung, auf die Daten der Session von überall in deiner Anwendung zuzugreifen, ohne sie als Props weitergeben zu müssen:
+
+```tsx:pages/users/[id].tsx
+import { useSession } from "next-auth/react";
+
+const User = () => {
+  const { data: session } = useSession();
+
+  if (!session) {
+    // Behandlung des nicht authentifizierten Zustands, z. B. Rendern eines SignIn-Komponenten
+    return <SignIn />;
+  }
+
+  return <p>Welcome {session.user.name}!</p>;
+};
+```
+
+## Einfügen von `user.id` in die Session
+
+`create-t3-app` ist so konfiguriert, dass es den [session callback](https://next-auth.js.org/configuration/callbacks#session-callback) in der NextAuth.js-Konfiguration verwendet, um die ID des Benutzers im `session`-Objekt einzufügen.
+
+```ts:pages/api/auth/[...nextauth].ts
+callbacks: {
+    session({ session, user }) {
+      if (session.user) {
+        session.user.id = user.id;
+      }
+      return session;
+    },
+  },
+```
+
+Dies ist mit einer Typdeklarationsdatei gekoppelt, um sicherzustellen, dass die `user.id` beim Zugriff auf das `session`-Objekt richtig getyped wird. Erfahre mehr über [`"Module Augmentation"`](https://next-auth.js.org/getting-started/typescript#module-augmentation) in den Dokumentationen von NextAuth.js.
```

Comment:
```suggestion
Dies ist mit einer Typdeklarationsdatei gekoppelt, um sicherzustellen, dass die `user.id` beim Zugriff auf das `session`-Objekt richtig getypd wird. Erfahre mehr über [`"Module Augmentation"`](https://next-auth.js.org/getting-started/typescript#module-augmentation) in den Dokumentationen von NextAuth.js.
```

Reply:
I dont like that suggestion tbh. Feel free to prove me wrong tho